### PR TITLE
Requester update

### PR DESCRIPTION
**Pull request by shazbert**
## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Unit test: "https://httpstat.us/200?sleep=20000" end point to sleep before reply

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool 
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes

### CODE REVIEW

File: exchanges/request/request.go [hunk 1]
```diff
@@ -247,36 +248,50 @@ func (r *Requester) DoRequest(req *http.Request, method, path string, headers ma
 		log.Printf("%s exchange request path: %s requires rate limiter: %v", r.Name, path, r.RequiresRateLimiter())
 	}
 
-	resp, err := r.HTTPClient.Do(req)
+	var timeoutError error
+	for i := 0; i < 4; i++ { // 3 attempts for extra requests due to client timeout
```
Comment (thrasher-) [inline exchanges/request/request.go]: Create a request var called DefaultTimeoutRetryAttempts and set that to 3, in the for loop change it to i < DefaultTimeoutRetryAttempts +1

Engine refactor will include code to make this overridable and optional based on user preferences.

Comment (shazbert) [reply exchanges/request/request.go]: Okay will do

Comment (gloriousCode) [reply exchanges/request/request.go]: Could we go with `i <= X` rather than `i < (X +1)` ?

Comment (thrasher-) [reply exchanges/request/request.go]: Destinction of retry attempts val and primary request is better IMO.

Comment (shazbert) [reply exchanges/request/request.go]: Yeah if we pass in 0 then it won't do a request.

**Pull request merged**